csharp/src/Ice/IceSSL/SSLEngine.cs

```diff
@@ -260,12 +197,68 @@ internal void verifyPeer(ConnectionInfo info, string description)
         }
     }
 
+    internal SslClientAuthenticationOptions createClientAuthenticationOptions(
+        RemoteCertificateValidationCallback remoteCertificateValidationCallback,
+        string host)
+    {
+        return new SslClientAuthenticationOptions
+        {
+            ClientCertificates = _certs,
+            LocalCertificateSelectionCallback = (sender, targetHost, certs, remoteCertificate, acceptableIssuers) =>
+            {
+                if (certs == null || certs.Count == 0)
+                {
+                    return null;
+                }
+                else if (certs.Count == 1)
+                {
+                    return certs[0];
+                }
+
+                // Use the first certificate that match the acceptable issuers.
+                if (acceptableIssuers != null && acceptableIssuers.Length > 0)
+                {
+                    foreach (X509Certificate certificate in certs)
+                    {
+                        if (Array.IndexOf(acceptableIssuers, certificate.Issuer) != -1)
+                        {
+                            return certificate;
+                        }
+                    }
+                }
+                return certs[0];
+            },
+            RemoteCertificateValidationCallback = remoteCertificateValidationCallback,
+            TargetHost = host,
+            CertificateRevocationCheckMode = X509RevocationMode.NoCheck
```

Comment:
I we should still perform revocation checks. If the user doesn't want it they can use the programmatic api :)

Reply:
This was our previous default, for self-signed certs that don't configure CRL/OCSP you will get a status unknown. How should we deal with this?

Reply:
I would assume this is only ran if OCSP or CRL are specified by as being available.

Reply:
> I would assume this is only ran if OCSP or CRL are specified by as being available. Could be wrong though. I think I'd prefer to keep the property.

I checked and that is not the case. You get `RevocationStatusUnknown` when the cert doesn't provide CRL. 

See https://learn.microsoft.com/en-us/dotnet/api/system.security.cryptography.x509certificates.x509chainstatusflags?view=net-8.0
> Specifies that it is not possible to determine whether the certificate has been revoked. This can be due to the certificate revocation list (CRL) being offline or unavailable.

Reply:
I'll agree that CRL is somewhat outdated now and more niche. OCSP is still quite relevant though. Generally good servers use OCSP stapling so the client doesn't need to perform a request.

Reply:
Ice servers don't support OCSP stapling as far as I remember. We can probably add this but it is not straightforward on all platforms. I think we better discuss revocation checks for IceSSL in a separate issue. For now, this keeps the 3.7 behavior.

Reply:
Ok, let's keep the property with the current default.